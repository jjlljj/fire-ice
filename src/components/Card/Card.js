import React from 'react'
import './Card.css'

export const Card = ({ house }) => {
  const { Name, Founded, Seats, Titles, CoatOfArms, AncestralWeapons } = house
  return (
    <div>
      <h2>{ Name }</h2>
      <h3>Founded: {Founded}</h3>
      <ul>
        <li>Seats: {Seats}</li>
        <li>Titles: {Titles} </li>
        <li>Coat of Arms: {CoatOfArms}</li>
        <li>Ancestral Weapons: {AncestralWeapons}</li>
      </ul>
    </div>
  )
}

export default Card