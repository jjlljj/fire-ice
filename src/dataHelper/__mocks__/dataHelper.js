import { mockCleanedHouses, mockHouses } from '../../__mocks__/mockData'

export const fetchHouses = () => {
  fetch('mocked fetch houses')
  return mockHouses
}

export const cleanHouses = () => {
  return mockCleanedHouses
}

export const fetchSwornMembers = () => {
  fetch('mocked fetch sworn members')
  return "Allyria Dayne, Allyria Dayne, Allyria Dayne"
}