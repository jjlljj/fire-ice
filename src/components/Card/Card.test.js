/* eslint-disable */
import React from 'react'
import { shallow } from 'enzyme'
import Card from './Card'
import { mockCleanedHouse } from '../../__mocks__/mockData'

describe('Card', () => {
  let renderedComponent

  beforeEach(() => {
    renderedComponent = shallow(<Card house={ mockCleanedHouse }/>)
  })

  it('should match snapshot', () => {
    expect(renderedComponent).toMatchSnapshot()
  })

  it('should match snapshot when passed house members', () => {
    const houseWithMembers = { ...mockCleanedHouse, members: "Charlie, Dee, Dennis, Mac" }

    renderedComponent = shallow(<Card house={ houseWithMembers }/>)
    expect(renderedComponent).toMatchSnapshot()
  })

})