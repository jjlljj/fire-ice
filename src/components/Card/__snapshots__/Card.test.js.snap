// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    house={
        Object {
            "AncestralWeapons": "Lady Forlorn",
            "CoatOfArms": "Three black ravens in flight, holding three red hearts, on a white field(Argent, three ravens volant sable, each clutching in their claws a heart gules)",
            "Founded": "",
            "Name": "House Corbray of Heart's Home",
            "Seats": "Heart's Home",
            "Titles": "Lord of Heart's Home, King of the Fingers (historical)",
            "Words": "",
            "swornMembers": Array [
              "https://www.anapioficeandfire.com/api/characters/255",
              "https://www.anapioficeandfire.com/api/characters/256",
              "https://www.anapioficeandfire.com/api/characters/447",
              "https://www.anapioficeandfire.com/api/characters/654",
              "https://www.anapioficeandfire.com/api/characters/677",
              "https://www.anapioficeandfire.com/api/characters/681",
              "https://www.anapioficeandfire.com/api/characters/871",
              "https://www.anapioficeandfire.com/api/characters/1182",
            ],
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2>
          House Corbray of Heart's Home
</h2>,
        <h3>
          Founded: 
          
</h3>,
        <ul>
          <li>
                    Seats: 
                    Heart's Home
          </li>
          <li>
                    Titles: 
                    Lord of Heart's Home, King of the Fingers (historical)
                     
          </li>
          <li>
                    Coat of Arms: 
                    Three black ravens in flight, holding three red hearts, on a white field(Argent, three ravens volant sable, each clutching in their claws a heart gules)
          </li>
          <li>
                    Ancestral Weapons: 
                    Lady Forlorn
          </li>
</ul>,
      ],
      "className": "Card",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "House Corbray of Heart's Home",
        },
        "ref": null,
        "rendered": "House Corbray of Heart's Home",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Founded: ",
            "",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Founded: ",
          "",
        ],
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li>
              Seats: 
              Heart's Home
</li>,
            <li>
              Titles: 
              Lord of Heart's Home, King of the Fingers (historical)
               
</li>,
            <li>
              Coat of Arms: 
              Three black ravens in flight, holding three red hearts, on a white field(Argent, three ravens volant sable, each clutching in their claws a heart gules)
</li>,
            <li>
              Ancestral Weapons: 
              Lady Forlorn
</li>,
            undefined,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Seats: ",
                "Heart's Home",
              ],
            },
            "ref": null,
            "rendered": Array [
              "Seats: ",
              "Heart's Home",
            ],
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Titles: ",
                "Lord of Heart's Home, King of the Fingers (historical)",
                " ",
              ],
            },
            "ref": null,
            "rendered": Array [
              "Titles: ",
              "Lord of Heart's Home, King of the Fingers (historical)",
              " ",
            ],
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Coat of Arms: ",
                "Three black ravens in flight, holding three red hearts, on a white field(Argent, three ravens volant sable, each clutching in their claws a heart gules)",
              ],
            },
            "ref": null,
            "rendered": Array [
              "Coat of Arms: ",
              "Three black ravens in flight, holding three red hearts, on a white field(Argent, three ravens volant sable, each clutching in their claws a heart gules)",
            ],
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Ancestral Weapons: ",
                "Lady Forlorn",
              ],
            },
            "ref": null,
            "rendered": Array [
              "Ancestral Weapons: ",
              "Lady Forlorn",
            ],
            "type": "li",
          },
          undefined,
        ],
        "type": "ul",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            House Corbray of Heart's Home
</h2>,
          <h3>
            Founded: 
            
</h3>,
          <ul>
            <li>
                        Seats: 
                        Heart's Home
            </li>
            <li>
                        Titles: 
                        Lord of Heart's Home, King of the Fingers (historical)
                         
            </li>
            <li>
                        Coat of Arms: 
                        Three black ravens in flight, holding three red hearts, on a white field(Argent, three ravens volant sable, each clutching in their claws a heart gules)
            </li>
            <li>
                        Ancestral Weapons: 
                        Lady Forlorn
            </li>
</ul>,
        ],
        "className": "Card",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "House Corbray of Heart's Home",
          },
          "ref": null,
          "rendered": "House Corbray of Heart's Home",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Founded: ",
              "",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Founded: ",
            "",
          ],
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li>
                Seats: 
                Heart's Home
</li>,
              <li>
                Titles: 
                Lord of Heart's Home, King of the Fingers (historical)
                 
</li>,
              <li>
                Coat of Arms: 
                Three black ravens in flight, holding three red hearts, on a white field(Argent, three ravens volant sable, each clutching in their claws a heart gules)
</li>,
              <li>
                Ancestral Weapons: 
                Lady Forlorn
</li>,
              undefined,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Seats: ",
                  "Heart's Home",
                ],
              },
              "ref": null,
              "rendered": Array [
                "Seats: ",
                "Heart's Home",
              ],
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Titles: ",
                  "Lord of Heart's Home, King of the Fingers (historical)",
                  " ",
                ],
              },
              "ref": null,
              "rendered": Array [
                "Titles: ",
                "Lord of Heart's Home, King of the Fingers (historical)",
                " ",
              ],
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Coat of Arms: ",
                  "Three black ravens in flight, holding three red hearts, on a white field(Argent, three ravens volant sable, each clutching in their claws a heart gules)",
                ],
              },
              "ref": null,
              "rendered": Array [
                "Coat of Arms: ",
                "Three black ravens in flight, holding three red hearts, on a white field(Argent, three ravens volant sable, each clutching in their claws a heart gules)",
              ],
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Ancestral Weapons: ",
                  "Lady Forlorn",
                ],
              },
              "ref": null,
              "rendered": Array [
                "Ancestral Weapons: ",
                "Lady Forlorn",
              ],
              "type": "li",
            },
            undefined,
          ],
          "type": "ul",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Card should match snapshot when passed house members 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    house={
        Object {
            "AncestralWeapons": "Lady Forlorn",
            "CoatOfArms": "Three black ravens in flight, holding three red hearts, on a white field(Argent, three ravens volant sable, each clutching in their claws a heart gules)",
            "Founded": "",
            "Name": "House Corbray of Heart's Home",
            "Seats": "Heart's Home",
            "Titles": "Lord of Heart's Home, King of the Fingers (historical)",
            "Words": "",
            "members": "Charlie, Dee, Dennis, Mac",
            "swornMembers": Array [
              "https://www.anapioficeandfire.com/api/characters/255",
              "https://www.anapioficeandfire.com/api/characters/256",
              "https://www.anapioficeandfire.com/api/characters/447",
              "https://www.anapioficeandfire.com/api/characters/654",
              "https://www.anapioficeandfire.com/api/characters/677",
              "https://www.anapioficeandfire.com/api/characters/681",
              "https://www.anapioficeandfire.com/api/characters/871",
              "https://www.anapioficeandfire.com/api/characters/1182",
            ],
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2>
          House Corbray of Heart's Home
</h2>,
        <h3>
          Founded: 
          
</h3>,
        <ul>
          <li>
                    Seats: 
                    Heart's Home
          </li>
          <li>
                    Titles: 
                    Lord of Heart's Home, King of the Fingers (historical)
                     
          </li>
          <li>
                    Coat of Arms: 
                    Three black ravens in flight, holding three red hearts, on a white field(Argent, three ravens volant sable, each clutching in their claws a heart gules)
          </li>
          <li>
                    Ancestral Weapons: 
                    Lady Forlorn
          </li>
          <li>
                    Sworn Members: 
                    Charlie, Dee, Dennis, Mac
                     
          </li>
</ul>,
      ],
      "className": "Card",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "House Corbray of Heart's Home",
        },
        "ref": null,
        "rendered": "House Corbray of Heart's Home",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Founded: ",
            "",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Founded: ",
          "",
        ],
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li>
              Seats: 
              Heart's Home
</li>,
            <li>
              Titles: 
              Lord of Heart's Home, King of the Fingers (historical)
               
</li>,
            <li>
              Coat of Arms: 
              Three black ravens in flight, holding three red hearts, on a white field(Argent, three ravens volant sable, each clutching in their claws a heart gules)
</li>,
            <li>
              Ancestral Weapons: 
              Lady Forlorn
</li>,
            <li>
              Sworn Members: 
              Charlie, Dee, Dennis, Mac
               
</li>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Seats: ",
                "Heart's Home",
              ],
            },
            "ref": null,
            "rendered": Array [
              "Seats: ",
              "Heart's Home",
            ],
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Titles: ",
                "Lord of Heart's Home, King of the Fingers (historical)",
                " ",
              ],
            },
            "ref": null,
            "rendered": Array [
              "Titles: ",
              "Lord of Heart's Home, King of the Fingers (historical)",
              " ",
            ],
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Coat of Arms: ",
                "Three black ravens in flight, holding three red hearts, on a white field(Argent, three ravens volant sable, each clutching in their claws a heart gules)",
              ],
            },
            "ref": null,
            "rendered": Array [
              "Coat of Arms: ",
              "Three black ravens in flight, holding three red hearts, on a white field(Argent, three ravens volant sable, each clutching in their claws a heart gules)",
            ],
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Ancestral Weapons: ",
                "Lady Forlorn",
              ],
            },
            "ref": null,
            "rendered": Array [
              "Ancestral Weapons: ",
              "Lady Forlorn",
            ],
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Sworn Members: ",
                "Charlie, Dee, Dennis, Mac",
                " ",
              ],
            },
            "ref": null,
            "rendered": Array [
              "Sworn Members: ",
              "Charlie, Dee, Dennis, Mac",
              " ",
            ],
            "type": "li",
          },
        ],
        "type": "ul",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            House Corbray of Heart's Home
</h2>,
          <h3>
            Founded: 
            
</h3>,
          <ul>
            <li>
                        Seats: 
                        Heart's Home
            </li>
            <li>
                        Titles: 
                        Lord of Heart's Home, King of the Fingers (historical)
                         
            </li>
            <li>
                        Coat of Arms: 
                        Three black ravens in flight, holding three red hearts, on a white field(Argent, three ravens volant sable, each clutching in their claws a heart gules)
            </li>
            <li>
                        Ancestral Weapons: 
                        Lady Forlorn
            </li>
            <li>
                        Sworn Members: 
                        Charlie, Dee, Dennis, Mac
                         
            </li>
</ul>,
        ],
        "className": "Card",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "House Corbray of Heart's Home",
          },
          "ref": null,
          "rendered": "House Corbray of Heart's Home",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Founded: ",
              "",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Founded: ",
            "",
          ],
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li>
                Seats: 
                Heart's Home
</li>,
              <li>
                Titles: 
                Lord of Heart's Home, King of the Fingers (historical)
                 
</li>,
              <li>
                Coat of Arms: 
                Three black ravens in flight, holding three red hearts, on a white field(Argent, three ravens volant sable, each clutching in their claws a heart gules)
</li>,
              <li>
                Ancestral Weapons: 
                Lady Forlorn
</li>,
              <li>
                Sworn Members: 
                Charlie, Dee, Dennis, Mac
                 
</li>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Seats: ",
                  "Heart's Home",
                ],
              },
              "ref": null,
              "rendered": Array [
                "Seats: ",
                "Heart's Home",
              ],
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Titles: ",
                  "Lord of Heart's Home, King of the Fingers (historical)",
                  " ",
                ],
              },
              "ref": null,
              "rendered": Array [
                "Titles: ",
                "Lord of Heart's Home, King of the Fingers (historical)",
                " ",
              ],
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Coat of Arms: ",
                  "Three black ravens in flight, holding three red hearts, on a white field(Argent, three ravens volant sable, each clutching in their claws a heart gules)",
                ],
              },
              "ref": null,
              "rendered": Array [
                "Coat of Arms: ",
                "Three black ravens in flight, holding three red hearts, on a white field(Argent, three ravens volant sable, each clutching in their claws a heart gules)",
              ],
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Ancestral Weapons: ",
                  "Lady Forlorn",
                ],
              },
              "ref": null,
              "rendered": Array [
                "Ancestral Weapons: ",
                "Lady Forlorn",
              ],
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Sworn Members: ",
                  "Charlie, Dee, Dennis, Mac",
                  " ",
                ],
              },
              "ref": null,
              "rendered": Array [
                "Sworn Members: ",
                "Charlie, Dee, Dennis, Mac",
                " ",
              ],
              "type": "li",
            },
          ],
          "type": "ul",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
